Sources/MapboxMaps/Gestures/GestureHandlers/PanGestureHandler.swift

```diff
@@ -73,46 +96,55 @@ internal final class PanGestureHandler: GestureHandler, PanGestureHandlerProtoco
             // it without further dragging. This specific time interval
             // is just the result of manual tuning.
             let decelerationTimeout: TimeInterval = 1.0 / 30.0
-            guard let lastChangedDate = lastChangedDate,
+            guard !mapboxMap.pointIsAboveHorizon(touchLocation),
+                  let lastChangedDate = lastChangedDate,
                   dateProvider.now.timeIntervalSince(lastChangedDate) < decelerationTimeout else {
-                      previousTouchLocation = nil
-                      self.lastChangedDate = nil
-                      mapboxMap.dragEnd()
-                      delegate?.gestureEnded(for: .pan, willAnimate: false)
+                      endInteraction(willAnimate: false)
                       return
                   }
-            var previousDecelerationLocation = touchLocation
+            // If the gesture is potentially ending near the horizon, continually reset the
+            // touchLocation to 3/4 of the way to the bottom of the map while decelerating
+            // to avoid simulated interaction with the more sensitvie area near the horizon.
+            let height = mapboxMap.size.height
+            let initialDecelerationLocation = CGPoint(
+                x: touchLocation.x,
+                y: max(touchLocation.y, 3 * height / 4))
             cameraAnimationsManager.decelerate(
-                location: touchLocation,
-                velocity: gestureRecognizer.velocity(in: view),
+                location: initialDecelerationLocation,
+                velocity: clampTouchLocation(velocity, previousTouchLocation: .zero),
```

Comment:
Just would like to understand, why is .zero here not causing issues when pan direction is clamped to be vertical?

Reply:
Previously, to achieve horizontal or vertical clamping during fling, we'd clamp the simulated touchLocation. While working on this, I realized that we could achieve the same effect more elegantly by setting the velocity to 0 for the dimension where we don't want any movement.

I'm just reusing the same method that was originally named with the idea that it'd be used to clamp positions. Now that it's used to clamp positions during drag and velocities during fling, I can update the name to be more generic.

Reply:
Updated in 86f19cb to hopefully be a bit more clear